jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -3404,9 +3404,13 @@ private static MethodHandle validateArgumentsOfWhileLoop(MethodHandle predHandle
 		Class<?> bodyReturnType = bodyType.returnType;
 		
 		/* The signature of the body handle must be either (V, A...)V or (A...)void */
-		if ((void.class != bodyReturnType) && (bodyReturnType != bodyType.arguments[0])) {
+		if ((void.class != bodyReturnType) 
+				&& ((0 == bodyType.arguments.length) 
```

Comment:
```((0 == bodyType.arguments.length) ```
This seems like a good candidate for a package private method.  Maybe `MethodType.isNoArgs()` or `hasNoArguments()`?